### DISCUSSION TITLE
Enhancing Configuration Management with GUI Interface

### DISCUSSION THREAD

**Discussion by 0x11c11e** [assafelovic/gpt-researcher#656] [Ideas]
I believe one of the milestones in the project roadmap should be to convert the settings that are currently adjustable in `config.py` and sourced from the `.env` file to a more user-friendly GUI interface. This would greatly enhance usability and make it easier for users to adjust settings without diving into environment variables or configuration files.

Currently, the following settings are adjustable from the `.env` file, but they could benefit from a GUI interface for more intuitive management:

- Retriever
- Embedding Provider
- Similarity Threshold
- LLM Provider
- Ollama Base URL
- Fast LLM Model
- Smart LLM Model
- Fast Token Limit
- Smart Token Limit
- Browse Chunk Max Length
- Summary Token Limit
- Temperature
- User Agent
- Max Search Results Per Query
- Memory Backend
- Total Words
- Report Format
- Max Iterations
- Agent Role
- Scraper
- Max Subtopics
- Document Path

Implementing a GUI for these settings would make the configuration process more accessible, especially for users who are not comfortable with editing code or environment files.

**Comment by rwecho** [comment]
Under what circumstances do we need a GUI?

**Comment by 0x11c11e** [comment]
- Not all users are comfortable with editing environment variables or configuration files. 
- A GUI can enhance the overall user experience by providing a more organized and structured way to manage settings.
- As the number of configurable settings grows, managing them through a GUI can be more scalable than maintaining extensive documentation for config files. 
- Some users might have accessibility needs that are better served by a GUI.
- Error reduction

**Comment by ElishaKay** [comment]
Agreed.

Here's what we're planning for the next big release.
![Screen Shot 2024-07-13 at 20 41 16](https://github.com/user-attachments/assets/ad6abd65-7dbd-4919-846b-7da683655ada)

You can [clone the PR](https://github.com/assafelovic/gpt-researcher/pull/523) to play with it or wait for it to be merged.
